.circleci/config.yml

```diff
@@ -133,11 +132,12 @@ jobs:
       - run: mkdir build
       - run: |
           cd build && cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=On -DENABLE_PYTHON_BINDINGS=On \
-          -DCPACK_GENERATOR=DEB -DCPACK_PACKAGE_VERSION_SUFFIX="-0ubuntu1-$(lsb_release -sc)"
-      - run: make -C build -j4
+          -DCPACK_GENERATOR=DEB -DCPACK_PACKAGE_VERSION_SUFFIX="-0ubuntu1-$(lsb_release -sc)" -DENABLE_ADDRESS_SANITIZER=ON
+      - run: make -C build -j2
```

Comment:
why `-j2` ? does `-DENABLE_ADDRESS_SANITIZER=ON` have impact on memory usage while building ?

Reply:
its hard to tell. CI would get OOM killed when it was j4 but previously the debug build was j4 and didnt get killed. i think potentially CIs reserve capacity ebbs and flows and this causes us to go through hard times here and there. what i do know is that we are allocated 2 cpus in CI so j4 is abusive.

Reply:
> does -DENABLE_ADDRESS_SANITIZER=ON have impact on memory usage while building ?
yeah, it does as it additionally stores for each allocated byte whether it's valid or not(roughly)